aip/0217.md

```diff
@@ -0,0 +1,82 @@
+---
+aip:
+  id: 217
+  state: reviewing
+  created: 2019-08-26
+permalink: /217
+redirect_from:
+  - /0217
+---
+
+# Unreachable resources
+
+Occasionally, a user may ask for a list of resources, and some set of resources
+in the list are temporarily unavailable. For example, a user may ask to list
+resources across multiple locations, but one of those locations is temporarily
+down. In this situation, it is still desirable to give the user as much as
+possible, while indicating that something is missing.
+
+## Guidance
+
+If a method to retrieve data is capable of partially failing due to one or more
+resources being temporarily unreachable, the response message **must** include
+a field to indicate this:
+
+```proto
+// The request message for ListBooks.
+message ListBooksRequest {
+  // The publisher to list books from.
+  // To list books across all publishers, use "publishers/-".
+  string parent = 1;
+
+  // Other fields...
+}
+
+message ListBooksResponse {
+  // The books matching the request.
+  repeated Book books = 1;
+
+  // The next page token, if there are more books matching the
+  // request.
+  string next_page_token = 2;
+
+  // Unreachable resources. If any books that would match the
+  // request were not included (for example, because one publisher's
+  // API was down), the unreachable resources are listed here.
+  repeated string unreachable = 3;
+}
+```
+
+- The field **must** be a repeated string, and **should** be named
+  `unreachable`.
+- The field **must** be set to the names of the resources which are the cause
+  of the issue, such as the locations or resources that could not be reached.
```

Comment:
Additionally, I think we should be clear about how homogeneous or otherwise these might be.

Suppose we have `clusters/x/nodes/y/tasks/z` and we're listing tasks:

- Cluster A has nodes A1 and A2 which are fine, so we can list tasks in them
- Cluster B has nodes B1 and B2, but B2 is failing. We only include tasks in B1
- Cluster C is down entirely

Would the unreachable list be "clusters/b/nodes/b1" and "clusters/c", or should it just be "clusters/b" and "clusters/c"?